app/src/test/java/org/oppia/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -280,6 +287,34 @@ class StateFragmentLocalTest {
     }
   }
 
+  @Test
+  fun testStateFragment_submitTwoWrongAnswers_hintAvailable_prevState_hintNotAvailable() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughState1()
+      submitTwoWrongAnswers()
+      onView(withId(R.id.hint_bulb)).check(matches(isDisplayed()))
+      onView(withId(R.id.previous_state_navigation_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.hint_bulb)).check(matches(not(isDisplayed())))
+    }
+  }
+
+  @Test
+  fun testStateFragment_submitTwoWrongAnswers_prevState_nextState_hintAvailable() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughState1()
+      submitTwoWrongAnswers()
+      onView(withId(R.id.previous_state_navigation_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.state_recycler_view)).perform(scrollToViewType(NEXT_NAVIGATION_BUTTON))
+      onView(withId(R.id.next_state_navigation_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.hint_bulb)).check(matches(isDisplayed()))
```

Comment:
This doesn’t seem right. Why is the hint visible in the next state? Nothing has been done by the user to trigger that.

Reply:
What this test is doing is:
1. The user submits 2 wrong answer
2. Hint bulb is visible
3. A user going to prev state - where hint bulb is not visible (tested on second test cases in this or already)
4. Coming back to current state where user submit two wrong answer - so hint bulb should be visible again.

Reply:
I must have read this & the previous test incorrectly. This does look correct--thanks!